File: OtherComponents/MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/NewMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace gov.va.medora.mdo.domain.sm
{
    [Serializable]
    public class NewMessage
    {
        private MailParticipant _from;

        internal MailParticipant From
        {
            get { return _from; }
            set { _from = value; }
        }
        private MailParticipant _to;

        internal MailParticipant To
        {
            get { return _to; }
            set { _to = value; }
        }
        private MailParticipant _cc;

        internal MailParticipant Cc
        {
            get { return _cc; }
            set { _cc = value; }
        }
        private string _subject;

        public string Subject
        {
            get { return _subject; }
            set { _subject = value; }
        }
        private string _body;

        public string Body
        {
            get { return _body; }
            set { _body = value; }
        }
        private TriageGroup _triageGroup;

        public TriageGroup TriageGroup
        {
            get { return _triageGroup; }
            set { _triageGroup = value; }
        }
        private bool _draft;

        public bool Draft
        {
            get { return _draft; }
            set { _draft = value; }
        }
        private Int64 _attachmentId;

        public Int64 AttachmentId
        {
            get { return _attachmentId; }
            set { _attachmentId = value; }
        }
        private string _attachmentName;

        public string AttachmentName
        {
            get { return _attachmentName; }
            set { _attachmentName = value; }
        }
        private Int64 _messageCategoryTypeId;

        public Int64 MessageCategoryTypeId
        {
            get { return _messageCategoryTypeId; }
            set { _messageCategoryTypeId = value; }
        }
    }
}

File: OtherComponents/MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/trash/NewMessage.java
package gov.va.med.mhv.sm.model;

import gov.va.med.mhv.sm.enumeration.MessageCategoryTypeEnum;

import java.io.Serializable;



/**
 * 
 * This is used as a DTO for sending messages.
 * 
 * Because of the data model the Message object is
 * not as useful because the subject, to, and from
 * members are separated into different objects.
 * 
 * This is a useful way for the web tier to package up
 * Messages into a simple object for processing
 * 
 * 
 * @author vhalommccarw
 *
 */



public class NewMessage implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = -8542725951535311517L;
	private MailParticipant from;
	private MailParticipant to;
	private MailParticipant cc;
	private String subject;
	private String body;
	private TriageGroup triageGroup;
	private boolean draft;
	private Long attachmentId;
	private String attachmentName;
	private Long messageCategoryTypeId;
	
	
	public MailParticipant getFrom() {
		return from;
	}
	public void setFrom(MailParticipant from) {
		this.from = from;
	}
	public MailParticipant getTo() {
		return to;
	}
	public void setTo(MailParticipant to) {
		this.to = to;
	}
	public String getSubject() {
		return subject;
	}
	public void setSubject(String subject) {
		this.subject = subject;
	}
	public String getBody() {
		return body;
	}
	public void setBody(String body) {
		this.body = body;
	}
	public TriageGroup getTriageGroup() {
		return triageGroup;
	}
	public void setTriageGroup(TriageGroup triageGroup) {
		this.triageGroup = triageGroup;
	}
	public boolean isDraft() {
		return draft;
	}
	public void setDraft(boolean draft) {
		this.draft = draft;
	}
	public Long getAttachmentId() {
		return attachmentId;
	}
	public void setAttachmentId(Long attachmentId) {
		this.attachmentId = attachmentId;
	}
	public Long getMessageCategoryTypeId() {
		return messageCategoryTypeId;
	}
	public void setMessageCategoryTypeId(Long messageCategoryTypeId) {
		this.messageCategoryTypeId = messageCategoryTypeId;
	}
	
	public MailParticipant getCc() {
		return cc;
	}
	public void setCc(MailParticipant cc) {
		this.cc = cc;
	}
	public String getAttachmentName() {
		return attachmentName;
	}
	public void setAttachmentName(String attachmentName) {
		this.attachmentName = attachmentName;
	}
}
